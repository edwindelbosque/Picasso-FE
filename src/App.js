import React, { Component } from 'react';
import { getFiveColors } from "../src/apiCalls.js/apiCalls";
import LoginForm from '../src/containersmm/loginForm/login'

import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      arrayOfColors: []
    }
  }
  componentDidMount = async () => {
    this.getFiveColors();
     
  };

  getFiveColors = (colorsRequest, model = 'default') => {
    const url = "http://colormind.io/api/";
    const data = {
        model : model,
    }
    if (colorsRequest) {
        data.input = colorsRequest
    }
    var http = new XMLHttpRequest();
    http.onreadystatechange = async () => {
        if(http.readyState == 4 && http.status == 200) {
            var palettes = await JSON.parse(http.responseText).result;
            this.setState({arrayOfColors: palettes.map(palette => this.colorFormats(palette))},  () => console.log(this.state))

        }
    }
    http.open("POST", url, true);
    http.send(JSON.stringify(data));
  }
// This function is to get the info for each specific color
colorFormats = (rgbColors) => {
  const r = rgbColors[0]
  const g = rgbColors[1]
  const b = rgbColors[2]
  fetch(`https://www.thecolorapi.com/id?format=string&rgb=${r},${g},${b}`)
  .then ( response => response.json())
  .then( data => console.log(data))
  .then( data => data)
}

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
        <LoginForm />
      </div>
    );
  }

}

export default App;
