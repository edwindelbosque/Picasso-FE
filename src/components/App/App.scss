.App {
	p {
		font-size: 35px;
		color: #0445ff;
	}
}
