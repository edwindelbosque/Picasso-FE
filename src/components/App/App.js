import React, { Component } from 'react';
import './App.scss';
import NavBar from '../NavBar/NavBar';
import Footer from '../Footer/Footer';
import GetRandomColors from '../RandomColor/RandomColor.js';

class App extends Component {
	constructor() {
		super();
		this.state = {
			arrayOfColors: [],
			userName: '',
			currentCatalog: 0,
			userId: 0,
			catalogs: [],
			palettes: [],
			showSaveMenu: false
		};
	}

	updateArrayOfColors = colors => {
		this.setState({ arrayOfColors: colors });
	};

	closeSaveMenu = () => {
		this.setState({
			showSaveMenu: false
		});
	};

	openSaveMenu = () => {
		this.setState({
			showSaveMenu: true
		});
	};

	wipeUserData = () => {
		this.setState({
			arrayOfColors: [],
			userName: '',
			currentCatalog: 0,
			userId: 0,
			catalogs: [],
			palettes: []
		});
	};

	updateCurrentUser = (user, catalogs) => {
		const { firstName, id } = user;
		this.setState({ userName: firstName, userId: id, catalogs: catalogs });
	};

	updateCurrentCatalog = id => {
		this.setState({ currentCatalog: id });
	};

	resetCurrentCatalog = () => {
		this.setState({ currentCatalog: 0 });
	};

	render() {
		return (
			<div className='App'>
				<GetRandomColors
					arrayOfColors={this.state.arrayOfColors}
					updateArrayOfColors={this.updateArrayOfColors}
					userID={this.state.userId}
					currentCatalog={this.state.currentCatalog}
					closeSaveMenu={this.closeSaveMenu}
					openSaveMenu={this.openSaveMenu}
					showSaveMenu={this.state.showSaveMenu}
					catalogs={this.state.catalogs}
					resetCurrentCatalog={this.resetCurrentCatalog}
				/>
				<NavBar
					userName={this.state.userName}
					catalogs={this.state.catalogs}
					updateCurrentUser={this.updateCurrentUser}
					updateCurrentCatalog={this.updateCurrentCatalog}
					wipeUserData={this.wipeUserData}
				/>
				<Footer />
			</div>
		);
	}
}

export default App;
