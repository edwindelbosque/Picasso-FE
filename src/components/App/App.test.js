import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<App />);
	});

	it.skip('App should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
