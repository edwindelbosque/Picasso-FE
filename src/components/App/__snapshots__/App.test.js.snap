// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match snapshot 1`] = `
<div
  className="App"
>
  <NavBar />
</div>
`;
